File: app/src/main/java/com/projects/andreafranco/imagegalleryapp/loaders/ImageListLoader.java
package com.projects.andreafranco.imagegalleryapp.loaders;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.content.AsyncTaskLoader;
import android.text.TextUtils;

import com.projects.andreafranco.imagegalleryapp.models.Image;
import com.projects.andreafranco.imagegalleryapp.utils.HttpUtils;

import java.util.ArrayList;

public class ImageListLoader extends AsyncTaskLoader<ArrayList<Image>> {

    String mQuery;

    public ImageListLoader(@NonNull Context context, String query) {
        super(context);
        mQuery = query;
    }

    @Nullable
    @Override
    public ArrayList<Image> loadInBackground() {
        if (mQuery == null || TextUtils.isEmpty(mQuery)) {
            return null;
        }
        return HttpUtils.fetchImageListData(mQuery);
    }

    @Override
    protected void onStartLoading() {
        forceLoad();
    }
}

File: app/src/main/java/com/projects/andreafranco/imagegalleryapp/loaders/ImageListLoader.kt
package com.projects.andreafranco.imagegalleryapp.loaders

import android.content.Context
import android.support.v4.content.AsyncTaskLoader
import android.text.TextUtils

import com.projects.andreafranco.imagegalleryapp.models.Image
import com.projects.andreafranco.imagegalleryapp.utils.HttpUtils

import java.util.ArrayList

class ImageListLoader(context: Context, val mQuery: String?) : AsyncTaskLoader<ArrayList<Image>>(context) {

    override fun loadInBackground(): ArrayList<Image>? {
        return if (mQuery == null || TextUtils.isEmpty(mQuery)) {
            null
        } else HttpUtils.fetchImageListData(mQuery)
    }

    override fun onStartLoading() {
        forceLoad()
    }
}
